src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -1008,6 +1029,29 @@ def retrieve_token_for_user(self, username, tenant, resource):
             self.persist_cached_creds()
         return (token_entry[_TOKEN_ENTRY_TOKEN_TYPE], token_entry[_ACCESS_TOKEN], token_entry)
 
+    def retrieve_ssh_certificate_for_user(self, tenant, modulus, exponent, refresh_token):
+        """
+        This is added only for vmssh feature.
+        It is a temporary solution and will deprecate after adoption to MSAL completely.
+        """
+        from azure.cli.core._msal import SSHCertificateClientApplication
+        scopes = ["https://pas.windows.net/CheckMyAccess/Linux/user_impersonation"]
+        tenant = tenant or 'organizations'
+        authority = self._ctx.cloud.endpoints.active_directory + '/' + tenant
+        app = SSHCertificateClientApplication(_CLIENT_ID, authority=authority)
+        jwk = {
+            "kty": "RSA",
+            "n": modulus,
+            "e": exponent,
+            "kid": str(hash((modulus, exponent)))
```

Comment:
This shouldn't be using the built-in hash because the value changes between runs. Here's the code that I've modified it to locally to make it generate the same hash each run to use the same SSH cert out of the token cache.

```
        key_hash = hashlib.sha256()
        key_hash.update(modulus.encode('utf-8'))
        key_hash.update(exponent.encode('utf-8'))
        key_id = key_hash.hexdigest()

        jwk = {
            "kty": "RSA",
            "n": modulus,
            "e": exponent,
            "kid": key_id
        }
```

Reply:
Thanks @rlrossiter . Updated.